cms/cms_toolbars.py

```diff
@@ -322,6 +323,18 @@ def in_apphook(self):
                 from cms.views import details
                 return resolver.func != details
 
+    def in_apphook_root(self):
+        """
+        Returns True if the request is for a page handled by an apphook, but
+        is also the page it is attached to.
+        :return: Boolean
+        """
+        with force_language(get_language_from_request(self.request)):
+            page = applications_page_check(self.request)
+            if page:
+                return self.request.path == page.get_absolute_url()
```

Comment:
I guess I didn't realize that `request.current_page` is ultimately doing the same thing... and is cached, so, thanks! =)